### PR TITLE
chore(seer): Add timing metric to requests to Seer

### PR DESCRIPTION
**Pull request by lobsterkatie**
This wraps our calls to Seer with a timing metric, tagged with the endpoint being called.

### CODE REVIEW

File: src/sentry/seer/signed_seer_api.py [hunk 1]
```diff
@@ -21,13 +22,20 @@ def make_signed_seer_api_request(
...
-        path,
-        body=body,
-        headers={"content-type": "application/json;charset=utf-8", **auth_headers},
-        **timeout_options,
-    )
+    with metrics.timer(
+        "seer.request_to_seer",
+        tags={
+            # Pull off query params, if any
+            "endpoint": path.split("?")[0],
```
Comment (JoshFerge) [inline src/sentry/seer/signed_seer_api.py:29]: can do `urlparse(url).path` instead

Comment (JoshFerge) [inline src/sentry/seer/signed_seer_api.py:29]: should be able to use `urlparse(url1).path` which might be a tad safer

File: src/sentry/seer/signed_seer_api.py [hunk 2]
```diff
@@ -21,13 +22,20 @@ def make_signed_seer_api_request(
...
         timeout_options["timeout"] = timeout
 
-    return connection_pool.urlopen(
-        "POST",
-        path,
-        body=body,
-        headers={"content-type": "application/json;charset=utf-8", **auth_headers},
-        **timeout_options,
-    )
+    with metrics.timer(
```
Comment (JoshFerge) [inline src/sentry/seer/signed_seer_api.py:25]: let's make the sample rate 100% while we're in EA / still debugging this?

File: src/sentry/seer/signed_seer_api.py (hunk 1 above)
Comment (lobsterkatie) [reply src/sentry/seer/signed_seer_api.py]: Didn't know you could use `urlparse` when all you had was a path! Just tested it out and it seems to work, though, so I made the change.

File: src/sentry/seer/signed_seer_api.py (hunk 2 above)
Comment (lobsterkatie) [reply src/sentry/seer/signed_seer_api.py:26]: This metric applies to all requests to Seer, not just grouping. Is it worth making it configurable (i.e., making it so you can pass in a higher sample rate)? Or are we fine with collecting 100% for all Seer features?

Comment (JoshFerge) [reply src/sentry/seer/signed_seer_api.py:26]: i think its fine

**Review by JoshFerge — APPROVED** [review]

**Pull request merged**